### PR TITLE
Era resolution #178

### PR DESCRIPTION
**Pull request by justkawal**
## Any follow ups?
#180 

## Description
- A temporary fix to get `{"period": value, "phase": value}`.
- A issue is made to handle it in generic way: #183 
- Also improved `toHuman()` to handle `None` aka (`NoneOption`).

## Why?
#178 

## How?
- Done by manually extracting the era when making ChainDescription Types.

### CODE REVIEW

File: packages/polkadart_scale_codec/lib/src/core/codec.dart [hunk 1]
```diff
@@ -576,6 +632,9 @@ class Codec {
         return source.boolean();
       case Primitive.Str:
         return source.str();
+      // TODO: Resolve issue: #183 - Extrinsic Era
```
Comment (leonardocustodio) [inline packages/polkadart_scale_codec/lib/src/core/codec.dart:635]: Should we remove this?

File: packages/polkadart_scale_codec/lib/src/core/codec.dart [hunk 2]
```diff
@@ -646,6 +705,10 @@ class Codec {
       case Primitive.U256:
         encoder.u256(value);
         return;
+      // TODO: Resolve issue: #183 - Extrinsic Era
```
Comment (leonardocustodio) [inline packages/polkadart_scale_codec/lib/src/core/codec.dart:708]: Should we remove this?

File: packages/polkadart_scale_codec/lib/src/core/types.dart [hunk 3]
```diff
@@ -48,6 +48,10 @@ enum Primitive {
 
   /// Single Character
   Char,
+
+  /// Mocking Extrinsic Era as primitive for quick fix.
+  /// TODO: Issue: #183 : Remove this quick-fix once Extrinsic Era is implemented in Generic Way
```
Comment (leonardocustodio) [inline packages/polkadart_scale_codec/lib/src/core/types.dart:53]: Should we remove this?

**Review by leonardocustodio — APPROVED** [review]

**Review by Lohann — CHANGES_REQUESTED** [review]
I'm missing unit tests for this new introduced code, please at least test the `Primitive.ExtrinsicEra`

File: packages/polkadart_scale_codec/lib/src/core/types.dart [hunk 4]
```diff
@@ -48,6 +48,10 @@ enum Primitive {
...
+  ExtrinsicEra,
```
Comment (Lohann) [inline packages/polkadart_scale_codec/lib/src/core/types.dart:54]: I don't like this hack solution, big chances of we miss this out here, why we need a quick fix for this?

Comment (leonardocustodio) [reply packages/polkadart_scale_codec/lib/src/core/types.dart:54]: We kind of need this for testing ;)
Having tests to ensure it works and that when we refactor everything is correct should be good for us.

**Pull request closed without merging**